MATLAB `AcadosSimSolver` fails to compile: `sim_compile_interface.m` uses wrong include path (`sim_common.h` not found)

## Summary

Creating an `AcadosSimSolver` from the MATLAB interface fails while compiling the sim MEX interface, with `fatal error: 'acados/sim/sim_common.h' file not found`. The OCP interface (`AcadosOcpSolver`) compiles fine on the same install. The cause is that `sim_compile_interface.m` points the compiler at `-I<acados_dir>` instead of `-I<acados_dir>/include` (and omits the blasfeo/hpipm/external include dirs), whereas the sibling `ocp_compile_interface.m` already uses the correct paths. It looks like `sim_compile_interface.m` was not updated in parallel with `ocp_compile_interface.m`.

## Environment

- acados version: `v0.5.4-16-g573f49bb4`
- OS: macOS 26.5 (arm64 / Apple Silicon)
- MATLAB: R2026a, compiler "Xcode with Clang"
- Built from source; `ACADOS_INSTALL_DIR` set via `setup_acados_matlab.m`
- (Not macOS-specific — this is a plain include-path error in the MATLAB build script.)

## Steps to reproduce

Any model that constructs an `AcadosSimSolver` triggers it (e.g. the getting-started closed-loop example), or minimally:

```matlab
import casadi.*
x = SX.sym('x',1); u = SX.sym('u',1); xdot = SX.sym('xdot',1);
model = AcadosModel();
model.x = x; model.u = u; model.xdot = xdot;
model.f_expl_expr = -x + u;
model.f_impl_expr = model.f_expl_expr - xdot;
model.name = 'mini';

sim = AcadosSim(); sim.model = model;
sim.solver_options.integrator_type = 'IRK';
sim.solver_options.T = 0.1;
sim_solver = AcadosSimSolver(sim);   % <-- fails here
```

## Observed error

```
compiling <acados>/interfaces/acados_matlab_octave/sim_solve.c
Building with 'Xcode with Clang'.
sim_solve.c:36:10: fatal error: 'acados/sim/sim_common.h' file not found
   36 | #include "acados/sim/sim_common.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
Error in sim_compile_interface (line 85)
Error in AcadosSimSolver/compile_mex_sim_interface_if_needed (line 319)
Error in AcadosSimSolver (line 114)
```

## Root cause

In `interfaces/acados_matlab_octave/sim_compile_interface.m` the include flag is

```matlab
acados_include = ['-I' acados_folder];          % -> -I<acados>   (WRONG)
```

but the headers are installed under `<acados>/include/`, so `#include "acados/sim/sim_common.h"` cannot be resolved. The interface also does not pass the blasfeo / hpipm / external include directories, which `sim_common.h` pulls in transitively.

For comparison, `ocp_compile_interface.m` (which works) already uses:

```matlab
acados_include            = ['-I' fullfile(acados_folder, 'include')];
acados_interfaces_include = ['-I' fullfile(acados_folder, 'interfaces')];
external_include          = ['-I' fullfile(acados_folder, 'external')];
blasfeo_include           = ['-I' fullfile(acados_folder, 'include', 'blasfeo', 'include')];
hpipm_include             = ['-I' fullfile(acados_folder, 'include', 'hpipm', 'include')];
```

## Proposed fix

Make `sim_compile_interface.m` mirror `ocp_compile_interface.m`:

```diff
     % set paths
     acados_mex_folder = fullfile(acados_folder, 'interfaces', 'acados_matlab_octave');
-    acados_include = ['-I' acados_folder];
+    acados_include = ['-I' fullfile(acados_folder, 'include')];
     acados_interfaces_include = ['-I' fullfile(acados_folder, 'interfaces')];
+    external_include = ['-I' fullfile(acados_folder, 'external')];
+    blasfeo_include = ['-I' fullfile(acados_folder, 'include', 'blasfeo', 'include')];
+    hpipm_include = ['-I' fullfile(acados_folder, 'include', 'hpipm', 'include')];
     acados_lib_path = ['-L' fullfile(acados_folder, 'lib')];
```

and add the new include dirs to both `mex(...)` invocations:

```diff
         if is_octave()
-            mex(acados_include, acados_interfaces_include, acados_lib_path,...
+            mex(acados_include, acados_interfaces_include, external_include, blasfeo_include, hpipm_include, acados_lib_path,...
                 '-lacados', '-lhpipm', '-lblasfeo', mex_files{ii})
         else
             FLAGS = 'CFLAGS=$CFLAGS -std=c99';
-            mex(mex_flags, FLAGS, acados_include, acados_interfaces_include, acados_lib_path, ...
+            mex(mex_flags, FLAGS, acados_include, acados_interfaces_include, external_include, blasfeo_include, hpipm_include, acados_lib_path, ...
                 '-lacados', '-lhpipm', '-lblasfeo', mex_files{ii}, '-outdir', output_dir)
         end
```

## Verification

With the patch applied, `sim_solve.c` / `sim_get.c` compile and `AcadosSimSolver` builds and runs correctly (verified end-to-end with an IRK integrator on a pendulum-on-cart model). No other files needed changes.

I'm happy to open a PR with this change if useful.

--- a/interfaces/acados_matlab_octave/sim_compile_interface.m
+++ b/interfaces/acados_matlab_octave/sim_compile_interface.m
@@ -37,8 +37,11 @@ function sim_compile_interface(output_dir)
 
     % set paths
     acados_mex_folder = fullfile(acados_folder, 'interfaces', 'acados_matlab_octave');
-    acados_include = ['-I' acados_folder];
+    acados_include = ['-I' fullfile(acados_folder, 'include')];
     acados_interfaces_include = ['-I' fullfile(acados_folder, 'interfaces')];
+    external_include = ['-I' fullfile(acados_folder, 'external')];
+    blasfeo_include = ['-I' fullfile(acados_folder, 'include', 'blasfeo', 'include')];
+    hpipm_include = ['-I' fullfile(acados_folder, 'include', 'hpipm', 'include')];
     acados_lib_path = ['-L' fullfile(acados_folder, 'lib')];
 
     mex_names = { ...
@@ -78,11 +81,11 @@ function sim_compile_interface(output_dir)
         disp(['compiling ', mex_files{ii}])
         if is_octave()
     %        mkoctfile -p CFLAGS
-            mex(acados_include, acados_interfaces_include, acados_lib_path,...
+            mex(acados_include, acados_interfaces_include, external_include, blasfeo_include, hpipm_include, acados_lib_path,...
                 '-lacados', '-lhpipm', '-lblasfeo', mex_files{ii})
         else
             FLAGS = 'CFLAGS=$CFLAGS -std=c99';
-            mex(mex_flags, FLAGS, acados_include, acados_interfaces_include, acados_lib_path, ...
+            mex(mex_flags, FLAGS, acados_include, acados_interfaces_include, external_include, blasfeo_include, hpipm_include, acados_lib_path, ...
                 '-lacados', '-lhpipm', '-lblasfeo', mex_files{ii}, '-outdir', output_dir)
         end
     end

